pkg/cluster/storage.go

```diff
@@ -132,20 +132,21 @@ func (c *Cluster) getBootstrapKeyFromStorage(ctx context.Context, storageClient
 		return nil, false, nil
 	}
 	if len(bootstrapList) > 1 {
-		return nil, false, errors.New("found multiple bootstrap keys in storage")
+		logrus.Warn("found multiple bootstrap keys in storage")
 	}
-	bootstrapKV := bootstrapList[0]
-	// checking for empty string bootstrap key
-	switch string(bootstrapKV.Key) {
-	case emptyStringKey:
-		logrus.Warn("bootstrap data encrypted with empty string, deleting and resaving with token")
-		c.saveBootstrap = true
-		if err := storageClient.Delete(ctx, emptyStringKey, bootstrapKV.Modified); err != nil {
-			return nil, false, err
+	for _, bootstrapKV := range bootstrapList {
```

Comment:
This seems non-deterministic - the correct behavior depends on the empty string key coming up first, which isn't guaranteed. Can we get some tests for it perhaps, to make sure it handles all possible combinations of keys and key return orders?

Reply:
I think this is deterministic enough since there is only one case in the switch, but ideally we could avoid re-listing the bootstrap keys.

Reply:
It's OK now, originally it returned from the case within the loop, which meant that it would process differently depending on which key was enumerated first.